develop_tag.rb

```diff
@@ -0,0 +1,135 @@
+class DevelopTag
+
+  def update_develop_tag
+    if !list_all_tags.empty?
+      assign_new_dev_tag
+    else
+      assign_first_dev_tag
+    end
+  end
+
+  def list_all_tags
+    list_each_tag = `git tag`
+    list_each_tag = list_each_tag.split("\n")
+    return list_each_tag
+  end
+
+  def list_tag_environ_and_version_num
+    list_each_tag = list_all_tags
+    i=0
+    tag_id_list = []
+    while i < list_each_tag.length
+      tag_split = list_each_tag[i].split("-v")
+      tag_environ_type = tag_split[0]
+      tag_version_number = tag_split[1].split(".")
+      tag_version_number.each {|ver_num| ver_num.to_i}
+      tag_version_number.unshift(tag_environ_type)
+      tag_id_list.push(tag_version_number)
+      i+=1
+    end
+    return tag_id_list
+  end
+
+  def identify_dev_tags
+    tag_id_list = list_tag_environ_and_version_num
+    j=0
+    list_dev_tags = []
+    while j < tag_id_list.length do
```

Comment:
this while can be achieved with a select as well. Saves you from having to keep track of the index and makes your code less likely to have mistakes in it